app/assets/javascripts/darkswarm/services/cart.js.coffee

```diff
@@ -115,6 +115,12 @@ Darkswarm.factory 'Cart', (CurrentOrder, Variants, $timeout, $http, $modal, $roo
       @line_items = []
       localStorageService.clearAll() # One day this will have to be moar GRANULAR
 
+    check_last_finalised_item: =>
+      if @line_items_finalised.length == 1
+        Messages.error(t 'orders_cannot_remove_the_final_item')
+
+      @line_items_finalised.length == 1
+
```

Comment:
This might be more clear with a different name, and if we leave it up to the controller to display the Message. Coupled with the other suggested change in the controller, this could be: 

```suggestion
    has_one_line_item: =>
        @line_items_finalised.length == 1
```

It might not be totally necessary to have it as a method on Cart, but if we do keep it as a separate method, we don't use the instance variable outside the Cart class.